pkg/virt-controller/watch/snapshot/snapshot.go

```diff
@@ -647,6 +634,41 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 			} else {
 				updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, "Source not locked"))
 			}
+
+			online, err := source.Online()
+			if err != nil {
+				return err
+			}
+
+			if online {
+				indications := []snapshotv1.Indication{snapshotv1.VMSnapshotOnlineSnapshotIndication}
+
+				ga, err := source.GuestAgent()
+				if err != nil {
+					return err
+				}
+
+				if ga {
+					indications = append(indications, snapshotv1.VMSnapshotGuestAgentIndication)
+				} else {
+					indications = append(indications, snapshotv1.VMSnapshotNoGuestAgentIndication)
+				}
+
+				vmSnapshotCpy.Status.Indications = indications
+
+				frozen, err := source.Frozen()
+				if err != nil {
+					return err
+				}
+
+				if frozen {
+					updateSnapshotCondition(vmSnapshotCpy, newFreezingCondition(corev1.ConditionTrue, vmSnapshotSourceFrozen))
+				} else {
+					updateSnapshotCondition(vmSnapshotCpy, newFreezingCondition(corev1.ConditionFalse, vmSnapshotSourceThawed))
+				}
+			} else {
+				updateSnapshotCondition(vmSnapshotCpy, newFreezingCondition(corev1.ConditionFalse, vmSnapshotSourceThawed))
```

Comment:
why adding the freezing condition if the snapshot is not online? although it is true that the source is thawed since we never freezed but seems irrelevant in such case

Reply:
yeah could go either way on this.  the ocd part of me wants to include everything but maybe it's confusing?